pkg/github/github.go

```diff
@@ -93,13 +117,102 @@ func NewClient(ctx context.Context, token string, opts ...Option) *GitHubClient
 	return g
 }
 
+// ListIssues lists all issues and returns their numbers in a repository matching the given criteria.
+func (g *GitHubClient) ListIssues(ctx context.Context, owner, repo string, labels []string, state string) ([]*Issue, error) {
+	backoff := retry.NewFibonacci(g.cfg.initialRetryDelay)
+	backoff = retry.WithMaxRetries(g.cfg.maxRetries, backoff)
+	backoff = retry.WithCappedDuration(g.cfg.maxRetryDelay, backoff)
+
+	var response []*Issue
+
+	if err := retry.Do(ctx, backoff, func(ctx context.Context) error {
+		issues, resp, err := g.client.Issues.ListByRepo(ctx, owner, repo, &github.IssueListByRepoOptions{
+			Labels: labels,
+			State:  state,
+		})
+		if err != nil {
+			if _, ok := ignoredStatusCodes[resp.StatusCode]; !ok {
+				return retry.RetryableError(err)
+			}
+
+			return fmt.Errorf("failed to list issues: %w", err)
+		}
+
+		for _, i := range issues {
+			response = append(response, &Issue{Number: i.Number})
+		}
+		return nil
+	}); err != nil {
+		return nil, fmt.Errorf("failed to list issues after retries: %w", err)
+	}
+
+	return response, nil
+}
+
+// CreateIssue creates an issue.
+func (g *GitHubClient) CreateIssue(ctx context.Context, owner, repo, title, body string, assignees, labels []string) (*Issue, error) {
+	backoff := retry.NewFibonacci(g.cfg.initialRetryDelay)
```

Comment:
I think this entire block (sans the response) everywhere could be encapsulated in a helper function like `g.withRetries()` to make things cleaner.

Reply:
Done